pennylane/ops/qubit.py

```diff
@@ -2961,6 +2961,73 @@ def diagonalizing_gates(self):
         return [QubitUnitary(self.eigendecomposition["eigvec"].conj().T, wires=list(self.wires))]
 
 
+class Projector(Observable):
```

Comment:
@mariaschuld, curious to hear your thoughts on the name? I'm happy with it, although we have (in the past) stuck to the naming convention of `Qubit<thing>` to avoid confusion with `Fock<thing>`. So this would result in naming this `QubitProjector`.

Reply:
I would drop that convention, and rather have `<thing>` and `Fock<thing>`, given that 95% of the ecosystem never interacts with non-qubit methods...

Reply:
Agree 💯